cedar-policy-validator/src/schema/test_579.rs

```diff
@@ -0,0 +1,2183 @@
+    );
+}
+#[test]
+fn E3a1_human() {
+    assert_parse_error_human(
+        &a1_human(E3_human()),
+        &ExpectedErrorMessageBuilder::error("failed to resolve type: NS2::MyType")
+            .help("`NS2::MyType` has not been declared as an entity type")
+            // TODO: why don't we have an underline here? (Uncommenting this produces test failure)
+            //.exactly_one_underline("NS2::MyType")
+            .build(),
+    );
+}
+#[test]
+fn E3a2_human() {
+    assert_parse_error_human(
+        &a2_human(E3_human()),
+        &ExpectedErrorMessageBuilder::error("failed to resolve type: NS2::MyType")
+            .help("`NS2::MyType` has not been declared as an entity type")
+            // TODO: why don't we have an underline here? (Uncommenting this produces test failure)
+            //.exactly_one_underline("NS2::MyType")
+            .build(),
+    );
+}
+#[test]
+fn E3b1_human() {
+    assert_parse_error_human(
+        &b1_human(E3_human()),
+        &ExpectedErrorMessageBuilder::error("failed to resolve type: NS2::MyType")
+            .help("`NS2::MyType` has not been declared as an entity type")
+            // TODO: why don't we have an underline here? (Uncommenting this produces test failure)
+            //.exactly_one_underline("NS2::MyType")
+            .build(),
+    );
+}
+#[test]
+fn E3b2_human() {
+    assert_parse_error_human(
+        &b2_human(E3_human()),
+        &ExpectedErrorMessageBuilder::error("failed to resolve type: NS2::MyType")
+            .help("`NS2::MyType` has not been declared as an entity type")
+            // TODO: why don't we have an underline here? (Uncommenting this produces test failure)
+            //.exactly_one_underline("NS2::MyType")
+            .build(),
+    );
+}
+#[test]
+fn E3c_human() {
+    assert_parse_error_human(
+        &c_human(E3_human()),
+        &ExpectedErrorMessageBuilder::error("failed to resolve type: NS2::MyType")
+            .help("`NS2::MyType` has not been declared as an entity type")
+            // TODO: why don't we have an underline here? (Uncommenting this produces test failure)
+            //.exactly_one_underline("NS1::MyType")
+            .build(),
+    );
+}
+#[test]
+fn E3d1_human() {
+    assert_parses_successfully_human(&d1_human(E3_human()));
+}
+#[test]
+fn E3d2_human() {
+    // this is an error because we currently don't support `appliesTo { principal: [common-type] }`.
+    // The error message could be more clear, e.g., specialized to check whether
+    // the type that failed to resolve would have resolved to a common type if
+    // it were allowed to.
+    assert_parse_error_human(
+        &d2_human(E3_human()),
+        &ExpectedErrorMessageBuilder::error("failed to resolve type: NS2::MyType")
+            .help("`NS2::MyType` has not been declared as an entity type")
+            .build(),
+    );
+}
+
+// json versions
+#[test]
+fn A1a1_json() {
+    assert_parses_successfully_json(A1X1_json(a1_json()));
+}
```

Comment:
Might be nice to combine the human & json versions into a single test so we can easily see if there are cases where the two formats have different behavior.

Reply:
Completed this reorg